### PR TITLE
Fix glinter errors

### PR DESCRIPTION
**Pull request by mdboom**
tl;dr: This is a bunch of whitespace changes and linter ignores which should have no functional change other than: Correcting the markdown formatting in the generated metrics docs and preparing us for a time when Glean treats warnings as errors.

Details below:

This fixes a number of issues caught by the Glean linter or "glinter", which provides guidance about defining metrics.  This tool has run as part of the build for quite some time, but its "nits" have been treated as warnings, not errors.  (But given how noisy Gradle otherwise is, I think these warnings have largely been ignored, in fairness).  In a soon-to-be-released version of Glean, these warnings will become errors, so I'm getting these fixes in now so as to not break the build when that upgrade occurs.

The majority of these changes are related to whitespace and are things caught by the [yamllint](https://github.com/adrienverge/yamllint) tool underpinning glinter.

An important change to call out is the use of `|` rather than `>` for multi-line strings.  `>` converts all newlines to a single space, so is not suitable for strings that are formatted in markdown, such as Glean's `description`, since it will remove the paragraph breaks.

Aside from that, the following nits were found:

```
Sorry, Glean found some glinter nits:
WARNING: USER_LIFETIME_EXPIRATION: activation.activation_id: Metrics with `user` lifetime cannot have an expiration date. They live as long as the user profile does.
ERROR: BASELINE_PING: events.total_uri_count: The baseline ping is Glean-internal. User metrics should go into the 'metrics' ping or custom pings.
ERROR: BASELINE_PING: metrics.search_count: The baseline ping is Glean-internal. User metrics should go into the 'metrics' ping or custom pings.
ERROR: COMMON_PREFIX: pocket: Within category 'pocket', all metrics begin with prefix 'pocket_top_site'. Remove prefixes and (possibly) rename category.
ERROR: COMMON_PREFIX: user_specified_search_engines: Within category 'user_specified_search_engines', all metrics begin with prefix 'custom_engine'. Remove prefixes and (possibly) rename category.

Please fix the above nits to continue.
To disable a check, add a `no_lint` parameter with a list of check names to disable.
This parameter can appear with each individual metric, or at the top-level to affect the entire file.
```

These have all been "grandfathered in", by adding `no_lint` parameters where necessary to disable these checks.  Going forward for new content, these checks should be helpful to discourage things we now know aren't good practice.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by mdboom** [comment]
Oh, let me generate the markdown docs and include them here.  They aren't identical, but they are more correct given the `|` to `>` change.
